include/nfc/tnep/tag.rst

```diff
@@ -0,0 +1,141 @@
+.. _tnep_tag_readme:
+
+TNEP for tag device
+###################
+
+The tag library provides functionality to implement the Tag NDEF Exchange Protocol (TNEP) for an NFC tag.
+
+The tag library is used in the :ref:`nfc_tnep_tag` sample.
+
+Initialization
+**************
+
+Before using the tag library in a tag application, you must configure the NDEF message buffers and define the TNEP services.
+
+Configuring NDEF message buffers
+================================
+
+The tag library communicates with the tag using NDEF message buffers.
+The buffers are initialized all at the same time by :cpp:func:`nfc_tnep_tag_tx_msg_buffer_register`.
+
+To use the NDEF message buffers, you must configure the following elements:
+
+* Provide a pointer to two TX buffers - one for emulating the NDEF message and one for encoding the new NDEF message.
+* Set a maximum buffer size in bytes, which is valid for both buffers.
+
+The buffers must be linked with the tag layer in the application.
+
+When the polling device writes a new NDEF message to the tag buffer, the application must inform the tag library by calling :cpp:func:`nfc_tnep_tag_rx_msg_indicate`.
+
+Defining TNEP services
+======================
+
+TNEP services are capabilities and features provided to the adjacent upper layer.
```

Comment:
What does "adjacent upper layer" mean? Adjacent to what?

Reply:
Probably this means that services provides their features to application which is some kind of upper layer

Reply:
Rewrote the section a bit - please check.